From e9bd75fc7c63a79e49438121058a5a315101be8e Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Thu, 29 Aug 2013 16:27:37 +0800
Subject: [PATCH 245/602] ENGR00277248 BT: fix BT tethering failure

Before android4.3, tethering directly get interface instead of now getting
from Linkproperties's route. so add dhcpcd for bt-pan is mandatory which
will fill bt-pan's Linkproperties's content.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 hdmidongle_6dq/init.rc | 8 ++++++++
 imx5x/init.rc          | 8 ++++++++
 imx6/etc/init.rc       | 8 ++++++++
 3 files changed, 24 insertions(+)

diff --git a/hdmidongle_6dq/init.rc b/hdmidongle_6dq/init.rc
index a52334a..6159134 100755
--- a/hdmidongle_6dq/init.rc
+++ b/hdmidongle_6dq/init.rc
@@ -116,6 +116,14 @@ service iprenew_p2p /system/bin/dhcpcd -n
     disabled
     oneshot
 
+service dhcpcd_bt-pan /system/bin/dhcpcd -BKLG
+    disabled
+    oneshot
+
+service iprenew_bt-pan /system/bin/dhcpcd -n
+    disabled
+    oneshot
+
 on fs
 # mount ext4 partitions
     mount_all /fstab.freescale
diff --git a/imx5x/init.rc b/imx5x/init.rc
index 4100d92..0dcd295 100644
--- a/imx5x/init.rc
+++ b/imx5x/init.rc
@@ -615,6 +615,14 @@ service dhcpcd_eth0 /system/bin/dhcpcd -AL
     disabled
     oneshot
 
+service dhcpcd_bt-pan /system/bin/dhcpcd -BKLG
+    disabled
+    oneshot
+
+service iprenew_bt-pan /system/bin/dhcpcd -n
+    disabled
+    oneshot
+
 service hostapd /system/bin/hostapd /data/misc/wifi/hostapd.conf
     socket hostapd_wlan0 dgram 660 root wifi
     user root
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 5406e54..750b868 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -653,6 +653,14 @@ service dhcpcd_eth0 /system/bin/dhcpcd -ABKL
     disabled
     oneshot
 
+service dhcpcd_bt-pan /system/bin/dhcpcd -BKLG
+    disabled
+    oneshot
+
+service iprenew_bt-pan /system/bin/dhcpcd -n
+    disabled
+    oneshot
+
 service hostapd /system/bin/hostapd /data/misc/wifi/hostapd.conf
     socket hostapd_wlan0 dgram 660 root wifi
     user root
-- 
1.8.0

